contrib/scripts/install_prereq

```diff
@@ -194,7 +194,13 @@ check_installed_debs() {
 	done
 	pkgs=$(aptitude -F '%c %p' search ${tocheck} 2>/dev/null | awk '/^p/{print $2}')
 	if [ ${#pkgs} -ne 0 ]; then
-		echo $pkgs | sed -r -e "s/ ?[^ :]+:i386//g"
+		P1=`echo $pkgs | sed -r -e "s/ ?[^ :]+:i386//g"`
+		DEB_ARCH=`uname -m`
+		if [ "${DEB_ARCH}" = "aarch64" ]; then
+			echo $P1 | sed -r -e "s/ ?[^ :]+:armhf//g"
+		else
+			echo $P1
+		fi
 	fi
```

Comment:
I believe all of this can be replaced with:

```
aptitude -F '%c %p' search $tocheck 2>/dev/null | awk '/^p/{print $2}' | grep -vE ':(i386|armhf)$'
```

Reply:
It might be confusing due to context, but I am proposing that `check_installed_debs` be changed to:

```
check_installed_debs() {
    for pack in "$@" ; do
        tocheck="${tocheck} ^${pack}$ ~P^${pack}$"
    done
    aptitude -F '%c %p' search $tocheck 2>/dev/null | awk '/^p/{print $2}' | grep -vE ':(i386|armhf)$'
}
```

Reply:
If you do that, then I believe you'll end up suppressing armhf packages on armhf platforms when they should properly be installed.

Reply:
Specifically, you only don't want armhf packages (arm7) on aarch64 (arm8) systems.

Reply:
> If you do that, then I believe you'll end up suppressing armhf packages on armhf platforms when they should properly be installed.

Would those packages be suffixed with `:armhf`? I feel like just scrubbing any package with a name that contains a colon would be sufficient for most case.

Reply:
I don't know. I was trying to be as minimally impacting on what looks like a pretty old, legacy script. I don't have the infrastructure to test it on a bunch of stuff. I was just trying to fix it so that Asterisk will install on aarch64/arm8 platforms correctly.

Reply:
I understand. I think excluding any package with a `:` in the name is sufficient.

Reply:
> Specifically, you only don't want armhf packages (arm7) on aarch64 (arm8) systems.

Please see https://github.com/raspberrypi/firmware/issues/1795